src/components/NewObject.vue

```diff
@@ -74,8 +75,21 @@ export default {
         })
         .then(response => {
           this.$eventBus.emit("displayMessage", { message: `New ${this.objectType} created`, status: "success" });
+          let _name = "";
+          if (response.data.root_type == "entity")
+          {
+            _name = "EntityDetails";
+          }
+          else if (response.data.root_type == "indicator")
+          {
+            _name = "IndicatorDetails";
+          }
+          else if (response.data.root_type == "observable")
+          {
+            _name = "ObservableDetails";
+          }
           this.$router.push({
-            name: response.data.root_type == "entity" ? "EntityDetails" : "IndicatorDetails",
+            name: _name,
```

Comment:
We can probably forward this to `"/<root_type>/<id>"` 

```javascript
router.push({ path: `/${response.data.root_type}/${response.data.id}` })
```

Reply:
I'll test and if it works I'll switch to this prettier solution.

Reply:
fixed in 667a154b734da070ae12851bb27672c5fc19d4e8